pyccel/parser/semantic.py

```diff
@@ -770,6 +770,7 @@ def _infere_type(self, expr, **settings):
     def _visit(self, expr, **settings):
         """Annotates the AST.
 
+        # TODO: Why is IndexedVariable mentioned here?
         IndexedVariable atoms are only used to manipulate expressions, we then,
         always have a Variable in the namespace."""
```

Comment:
Given that `IndexedVariable` will not exist anymore after this PR, I think that we should understand this comment (if possible) and then update it or remove it!

@saidctb Do you have any clue?

Reply:
Ah! I had forgotten about that. That definitely needs fixing before we merge. It seems like a message that would make sense in the `insert_variable` function but not in the `_visit` function. I suspect it can be removed in any case

Reply:
I have no idea about that comment

Reply:
Let's remove it then!

Reply:
Git blame says it was written by Ahmed on 18/11/2018. Does everyone agree that it can be removed?

Reply:
yes you can remove it